controllers/apps/apimanager_managed_routes_handler.go

```diff
@@ -0,0 +1,137 @@
+package controllers
+
+import (
+	"context"
+
+	appscommon "github.com/3scale/3scale-operator/apis/apps"
+	appsv1alpha1 "github.com/3scale/3scale-operator/apis/apps/v1alpha1"
+	"github.com/3scale/3scale-operator/pkg/3scale/amp/component"
+	"github.com/go-logr/logr"
+	appsv1 "github.com/openshift/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ handler.EventHandler = &APIManagerRoutesHandler{}
+
+// APIManagerRoutesHandler is an EventHandler that maps an existing OpenShift
+// route to an APIManager. This handler should only be used on Route objects
+// and when APIManager is used.
+type APIManagerRoutesHandler struct {
+	k8sclient client.Client
+	logger    logr.Logger
+}
+
+// 	// Create is called in response to an create event - e.g. Pod Creation.
+func (h *APIManagerRoutesHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	apimanagerReconcileRequest := h.getAPIManagerOwnerReconcileRequest(evt.Meta)
+	if apimanagerReconcileRequest != nil {
+		q.Add(*apimanagerReconcileRequest)
+	}
+}
+
+// 	// Update is called in response to an update event -  e.g. Pod Updated.
+func (h *APIManagerRoutesHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	apimanagerReconcileRequestOld := h.getAPIManagerOwnerReconcileRequest(evt.MetaOld)
+	if apimanagerReconcileRequestOld != nil {
+		q.Add(*apimanagerReconcileRequestOld)
+	}
+	apimanagerReconcileRequestNew := h.getAPIManagerOwnerReconcileRequest(evt.MetaNew)
+	if apimanagerReconcileRequestNew != nil {
+		q.Add(*apimanagerReconcileRequestNew)
+	}
+}
+
+// 	// Delete is called in response to a delete event - e.g. Pod Deleted.
+func (h *APIManagerRoutesHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	apimanagerReconcileRequest := h.getAPIManagerOwnerReconcileRequest(evt.Meta)
+	if apimanagerReconcileRequest != nil {
+		q.Add(*apimanagerReconcileRequest)
+	}
+}
+
+// 	// Generic is called in response to an event of an unknown type or a synthetic event triggered as a cron or
+// 	// external trigger request - e.g. reconcile Autoscaling, or a Webhook.
+func (h *APIManagerRoutesHandler) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	apimanagerReconcileRequest := h.getAPIManagerOwnerReconcileRequest(evt.Meta)
+	if apimanagerReconcileRequest != nil {
+		q.Add(*apimanagerReconcileRequest)
+	}
+}
+
+func (h *APIManagerRoutesHandler) getAPIManagerOwnerReconcileRequest(object metav1.Object) *reconcile.Request {
+	if object == nil {
+		return nil
+	}
+	h.logger.V(2).Info("Processing meta object", "Name", object.GetName(), "Namespace", object.GetNamespace())
```

Comment:
why `V(2)`? Do we have a way to filter out `V(2)` and let `V(1)`. As I understand `h.logger.Info` is info and `h.logger.V(X).info` is debug

Reply:
Yes, the way to filter out v(2) and v(1) is through the use of --zap-log-level.
Debug is V(1).info.

I don't want to have the same level of verbosity as debug (1), I prefer it to be even on a more verbose level as this happens for each route change event